### PR TITLE
feat: [IDP-1315]: GitHub Actions data points Support

### PR DESCRIPTION
**Pull request by vigneswara-propelo**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)

### CODE REVIEW

File: idp-service/src/main/resources/migrations/scorecard/checks.json [hunk 1]
```diff
@@ -489,5 +489,25 @@
     "tags": ["Software quality", "Development maturity"],
     "defaultBehaviour": "FAIL",
     "failMessage": "Gitleaks file does not exist"
+  },
```
Comment (sathish-soundarapandian) [inline idp-service/src/main/resources/migrations/scorecard/checks.json:492]: We will be adding other checks for Github actions later ?

**Review by vikyathharekal — COMMENTED** [review]
> Reviewed 14 of 14 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)* status: 14 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/resources/migrations/scorecard/checks.json` line 492 at r1](https://reviewable.io/reviews/harness/harness-core/53998#gh-1354026567:-NgS3RYi8jyq7PfvG4Oh:b32v57) ([raw file](https://github.com/harness/harness-core/blob/d563a723f1c359a14e3127beab15e3950bdc5570/idp-service/src/main/resources/migrations/scorecard/checks.json#L492)):*
<details><summary><i>Previously, sathish-soundarapandian (Sathish) wrote…</i></summary><blockquote>

We will be adding other checks for Github actions later ?
</blockquote></details>

+1
___
*[`idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json` line 329 at r1](https://reviewable.io/reviews/harness/harness-core/53998#-NgS44vM1yGWfgPX9PDP:-NgS44vM1yGWfgPX9PDQ:b-bf23gx) ([raw file](https://github.com/harness/harness-core/blob/d563a723f1c359a14e3127beab15e3950bdc5570/idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json#L329)):*
> ```JSON
>     "apiRequestDetails": {
>       "method": "GET",
>       "url": "{API_BASE_URL}/repos/{REPOSITORY_OWNER}/{REPOSITORY_NAME}/actions/workflows",
> ```

nit: We have `https://{API_BASE_URL}` for gitlab. But just `{API_BASE_URL}`. Why is it different? Can we make it consistent?



_Code quote:_
```JSON
{API_BASE_URL}
```

**Review by vigneswara-propelo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)* status: 14 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/resources/migrations/scorecard/checks.json` line 492 at r1](https://reviewable.io/reviews/harness/harness-core/53998#gh-1354026567:-NgSZPHh0xYMOdesIoA4:b-zes1b5) ([raw file](https://github.com/harness/harness-core/blob/d563a723f1c359a14e3127beab15e3950bdc5570/idp-service/src/main/resources/migrations/scorecard/checks.json#L492)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

+1
</blockquote></details>

For the other two data points, we need WORKFLOW_ID to be mentioned in the conditional Input.
Since, this workflow id or filename vary based on customer choice we cannot determine any default value for this. Therefore, i have not created any default checks for them
___
*[`idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json` line 329 at r1](https://reviewable.io/reviews/harness/harness-core/53998#-NgS44vM1yGWfgPX9PDP:-NgSY3wID9v1GXLB6T9M:b-42g9ts) ([raw file](https://github.com/harness/harness-core/blob/d563a723f1c359a14e3127beab15e3950bdc5570/idp-service/src/main/resources/migrations/scorecard/datasourceLocations.json#L329)):*
<details><summary><i>Previously, vikyathharekal (Vikyath Harekal) wrote…</i></summary><blockquote>

nit: We have `https://{API_BASE_URL}` for gitlab. But just `{API_BASE_URL}`. Why is it different? Can we make it consistent?
</blockquote></details>

The reason why there is difference in Github and Gitlab is that the base URLs are little different & for gitlab we need to truncate 'v4' in the apiBaseUrl. Hence, I have made use of host itself. 
GitHub -> apiBaseUrl
GitLab -> host

**Review by vigneswara-propelo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)* status: 14 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`idp-service/src/main/resources/migrations/scorecard/checks.json` line 492 at r1](https://reviewable.io/reviews/harness/harness-core/53998#gh-1354026567:-NgSaRHw9GH-W2sI0o-L:b-ik5pbc) ([raw file](https://github.com/harness/harness-core/blob/d563a723f1c359a14e3127beab15e3950bdc5570/idp-service/src/main/resources/migrations/scorecard/checks.json#L492)):*
<details><summary><i>Previously, vigneswara-propelo (Vigneswara) wrote…</i></summary><blockquote>

For the other two data points, we need WORKFLOW_ID to be mentioned in the conditional Input.
Since, this workflow id or filename vary based on customer choice we cannot determine any default value for this. Therefore, i have not created any default checks for them
</blockquote></details>

CC: @sathish-soundarapandian

**Review by vikyathharekal — DISMISSED** [review]
> Reviewed 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)* status: 16 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by sathish-soundarapandian — APPROVED** [review]

**Comment by vigneswara-propelo** [comment]
trigger uts

**Review by vikyathharekal — APPROVED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53998)* status: 19 file(s) to review. 1. unresolved discussions.

**Comment by vigneswara-propelo** [comment]
trigger ut7, ut8

**Comment by vigneswara-propelo** [comment]
trigger review

**Pull request merged**